storage/statedb/cache_fastcache.go

```diff
@@ -52,30 +62,64 @@ func NewFastCache(config *TrieNodeCacheConfig) TrieNodeCache {
 		return nil
 	}
 
-	fc := &FastCache{fast: fastcache.LoadFromFileOrNew(config.FastCacheFileDir, config.LocalCacheSizeMB*1024*1024)} // Convert MB to Byte
-	stats := fc.UpdateStats().(fastcache.Stats)
+	fc := &FastCache{new: fastcache.New(config.LocalCacheSizeMB * 1024 * 1024)} // Convert MB to Byte
+	go fc.loadFromBackground(config)
 
-	logger.Info("Initialize local trie node cache (fastCache)",
-		"MaxMB", config.LocalCacheSizeMB, "FilePath", config.FastCacheFileDir,
-		"LoadedBytes", stats.BytesSize, "LoadedEntries", stats.EntriesCount)
+	fmt.Println("Initialize local trie node cache (fastCache)",
+		"MaxMB", config.LocalCacheSizeMB, "FilePath", config.FastCacheFileDir)
 	return fc
 }
 
+func (cache *FastCache) loadFromBackground(config *TrieNodeCacheConfig) {
+	start := time.Now()
+	cache.old = fastcache.LoadFromFileOrNew(config.FastCacheFileDir, config.LocalCacheSizeMB*1024*1024)
+	stats := cache.UpdateStats().(fastcache.Stats)
+	// if there is no saved cache, use only new cache
+	if stats.EntriesCount == 0 {
+		logger.Info("There is no saved cache in the fastcache directory",
+			"FilePath", config.FastCacheFileDir)
+		cache.old = nil
+		return
+	}
+	// if we loaded the saved cache successfully, reset new cache to reclaim the memory
+	cache.new.Reset()
+	fmt.Println("Finished loading fastcache from background", "elapsed", time.Since(start),
+		"LoadedBytes", stats.BytesSize, "LoadedEntries", stats.EntriesCount)
+}
+
 func (cache *FastCache) Get(k []byte) []byte {
-	return cache.fast.Get(nil, k)
+	if cache.old != nil {
+		if data := cache.old.Get(nil, k); data != nil {
+			return data
+		}
+	}
+	return cache.new.Get(nil, k)
 }
 
 func (cache *FastCache) Set(k, v []byte) {
-	cache.fast.Set(k, v)
+	if cache.old != nil {
```

Comment:
`cache.old` is used to check if background loading has finished or not and can be accessed from different goroutines concurrently.